src/prefect/environments/storage/github.py

```diff
@@ -0,0 +1,131 @@
+from typing import TYPE_CHECKING, Any, Dict, List
+
+
+from prefect.environments.storage import Storage
+from prefect.utilities.storage import extract_flow_from_file
+
+if TYPE_CHECKING:
+    from prefect.core.flow import Flow
+
+
+class GitHub(Storage):
+    """
+    GitHub storage class. This class represents the Storage interface for Flows stored
+    in `.py` files in a GitHub repository.
+
+    This class represents a mapping of flow name to file paths contained in the git repo,
+    meaning that all flow files should be pushed independently. A typical workflow using
+    this storage type might look like the following:
+
+    Compose flow `.py` file where flow has GitHub storage:
```

Comment:
Not sure how this will render in our docs, but perhaps make these steps bullets or numbered?